fsw/cfe-core/src/sb/cfe_sb_msg_id_util.c

```diff
@@ -185,3 +185,27 @@ void CFE_SB_SetMsgId(CFE_SB_MsgPtr_t MsgPtr,
 
 #endif 
 }/* end CFE_SB_SetMsgId */
+
+/*
+ * Function: CFE_SB_GetPktType - See API and header file for details
+ */
+uint8 CFE_SB_GetPktType(CFE_SB_MsgId_t MsgId)
+{
+
+  CFE_SB_MsgId_Atom_t Val = MsgId;
+  uint8 PktType;
+
+  if (CFE_SB_IsValidMsgId(MsgId))
+  {
+#ifndef MESSAGE_FORMAT_IS_CCSDS_VER_2
+    PktType = (CFE_TST(Val,12) == true) ? CFE_SB_PKTTYPE_CMD : CFE_SB_PKTTYPE_TLM;
```

Comment:
The extra `== true` here is redundant as the `CFE_TST()` macro already results in a boolean.  But it is potentially problematic if the definition of `CFE_TST()` macro changes as it only tests for one possible `true` value.  Just testing the value of `CFE_TST(Val,12)` directly is a little more future proof.

This is also (I believe) the only use of ternary conditional operator in our runtime codebase.  Not that I have a problem with it, but I think it was avoided out of concern that some compilers might not support it.  AFAIK it is standard since C99, but there is a (slight) risk that some compiler might have an issue with it.